### PR TITLE
Move task queue to memory

### PR DESCRIPTION
**Pull request by jssblck**
# Overview

We initially designed the task queue to persist to disk, with the intention that it'd survive restarts of Broker- primarily so that if Broker crashes, it can pick up where it left off.

We've gotten lots of support tickets that indicate that this functionality is confusing and unhelpful- especially given that Broker is also designed to mitigate the impact of queue deletion; a common troubleshooting tactic users have reported using is simply deleting the task queue files on disk.

This PR updates the task queue to simply exist in memory alone. When Broker shuts down, the queue is therefore cleared, and when Broker starts it constructs the queue from scratch.

Broker still avoids uploading duplicate revisions via the local database.

## Acceptance criteria

The task queue is in memory.

## Testing plan

I primarily relied on automated testing, which validates the task queue works. I also ran Broker in my local env and validated things still work as expected:

```
; cargo run -- run
   Compiling broker v0.2.3 (/Users/jessica/projects/broker)
    Finished dev [unoptimized + debuginfo] target(s) in 2.11s
     Running `target/debug/broker run`
2023-07-21T01:40:27.613672Z Debug artifacts being stored in '/Users/jessica/.config/fossa/broker/debug/trace'
2023-07-21T01:40:27.630424Z Polling 'git::ssh::git@github.com:replit/upm.git'
2023-07-21T01:40:33.179372Z Enqueued task to scan 'git::ssh::git@github.com:replit/upm.git' at 'git::branch::main@f2aecc45e5b53b0d269953d628bf6f9979904f3e'
2023-07-21T01:40:33.179424Z Next poll interval for 'git::ssh::git@github.com:replit/upm.git' in 3600s
2023-07-21T01:40:33.179737Z Scanning 'git::ssh::git@github.com:replit/upm.git' at 'git::branch::main@f2aecc45e5b53b0d269953d628bf6f9979904f3e'
2023-07-21T01:40:37.650129Z Scanned 'git::ssh::git@github.com:replit/upm.git' at 'git::branch::main@f2aecc45e5b53b0d269953d628bf6f9979904f3e'
2023-07-21T01:40:37.658973Z Uploading scan for project: 'git@github.com:replit/upm.git@f2aecc45e5b53b0d269953d628bf6f9979904f3e (main)'
2023-07-21T01:40:37.971824Z Uploaded scan for project 'git@github.com:replit/upm.git@f2aecc45e5b53b0d269953d628bf6f9979904f3e (main)' as locator: 'custom+24357/git@github.com:replit/upm.git$f2aecc45e5b53b0d269953d628bf6f9979904f3e'
```

## Risks

The differences are:

* Tasks are now cleared when Broker shuts down.
* Task queues now must have a limit (and therefore backpressure) to avoid infinitely growing queues.

I think neither of these changes are risky. The backpressure problem is automatically managed by the task queue mechanism, and tasks will simply be re-created when Broker starts back up. 

The only real potential for task re-work is that if Broker _clones and analyzes a project_, but _does not upload it_, when it restarts it'll need to clone and analyze again. I don't think this is a major cost to reduce confusion for end users, and could be mitigated by e.g. caching analysis results in the database if we wanted to (we'd store by `(integration, cli_version)` tuples).

## Checklist

- [x] I added tests for this PR's change (or explained in the PR description why tests don't make sense).

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -62,13 +62,14 @@ tracing-test = { version = "0.2.4", features = ["no-env-filter"] }
...
 regex = "1.8.4"
-srclib = { git = "https://github.com/fossas/foundation-libs", branch = "add-srclib" }
+srclib = { git = "https://github.com/fossas/foundation-libs" }
 walkdir = "2.3.3"
 tar = "0.4.38"
 libflate = "1.3.0"
 typed-builder = "0.14.0"
 which = "4.4.0"
 tikv-jemallocator = { version = "0.5.0", optional = true }
+deadqueue = "0.2.4"
```
Comment (spatten) [inline Cargo.toml:72]: According to the deadqueue docs, we need to add the `limited` feature if we want to use it: https://docs.rs/deadqueue/latest/deadqueue/.

I'm surprised that it works without it 🤔

File: src/queue.rs [hunk 2]
```diff
@@ -38,297 +29,64 @@ pub enum Error {
...
-            If you are certain no other Broker instances are running, deleting the lock file may recover this error.
-            "})
-            .describe_lazy(|| formatdoc! {"
-            Queue working state is stored on disk, and relies on a lockfile to guard access.
-            For this particular queue, this lock file is located at '{}'.
-            ", lock_path.display()})
-            .map(Self::new)
+    /// Sends an item into the queue.
+    pub async fn send(&self, item: &T) -> Result<(), Report<Error>> {
+        let encoded = serde_json::to_vec(item).context(Error::Serialize)?;
```
Comment (spatten) [inline src/queue.rs:71]: Are the `Error::IO` and `Error::Open` errors used anywhere?

**Review by spatten — APPROVED** [review]
Two minor nits, but other than that looks good to me :+1:

File: Cargo.toml (hunk 1 above)
Comment (jssblck) [reply Cargo.toml:72]: It is actually there by default! https://docs.rs/crate/deadqueue/latest/features

File: src/queue.rs (hunk 2 above)
Comment (jssblck) [reply src/queue.rs:63]: Indeed they are not, removed. Good catch!

**Pull request merged**